### PR TITLE
mgr/dashboard: fix missing token expiration on load 

### PR DESCRIPTION
**Pull request by nizamial09**
when first viewing the page, the token expiration date is missing which gets loaded after a while. fixing this by handling the subscription better. If my assumption is correct, this issue started breaking the multi-cluster e2es after the carbon switch

Fixes: https://tracker.ceph.com/issues/67840

**BEFORE**

https://github.com/user-attachments/assets/94f1c647-b4e9-4a41-88db-d9d94044b66a

**AFTER**

https://github.com/user-attachments/assets/30c5b9ec-3fc5-4672-929f-683eb08b12ac

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/multi-cluster/multi-cluster-list/multi-cluster-list.component.ts [hunk 1]
```diff
@@ -162,15 +142,38 @@ export class MultiClusterListComponent extends ListWithDetails implements OnInit
...
+            this.checkClusterConnectionStatus();
+            this.data.forEach((cluster: any) => {
+              cluster['remainingTimeWithoutSeconds'] = 0;
+              if (cluster['ttl'] && cluster['ttl'] > 0) {
+                cluster['ttl'] = cluster['ttl'] * 1000;
+                cluster['remainingTimeWithoutSeconds'] = this.getRemainingTimeWithoutSeconds(
+                  cluster['ttl']
+                );
+                cluster['remainingDays'] = this.getRemainingDays(cluster['ttl']);
+              }
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/multi-cluster/multi-cluster-list/multi-cluster-list.component.ts:171]: lets break this into small readable functions which will help in improving readability of this code.

Thanks

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Comment by afreen23** [comment]
Hey @nizamial09 can we rebase and take this forward, thanks

**Comment by afreen23** [comment]
Hey @nizamial09 can we rebase and take this forward, thanks

**Pull request closed without merging**